vlib/v/ast/README.md

```diff
@@ -0,0 +1,2837 @@
+# Vlang abstract syntax tree(AST) introduction
+
+## Overview
+
+The Vlang abstract syntax tree is implemented by sum type. the following is declaration:
+
+```v
+pub type TypeDecl = AliasTypeDecl | FnTypeDecl | SumTypeDecl
+
+pub type Expr = AnonFn | ArrayInit | AsCast | Assoc | AtExpr | BoolLiteral | CTempVar |
+	CallExpr | CastExpr | ChanInit | CharLiteral | Comment | ComptimeCall | ConcatExpr | EnumVal |
+	FloatLiteral | Ident | IfExpr | IfGuardExpr | IndexExpr | InfixExpr | IntegerLiteral |
+	Likely | LockExpr | MapInit | MatchExpr | None | OrExpr | ParExpr | PostfixExpr | PrefixExpr |
+	RangeExpr | SelectExpr | SelectorExpr | SizeOf | SqlExpr | StringInterLiteral | StringLiteral |
+	StructInit | Type | TypeOf | UnsafeExpr
+
+pub type Stmt = AssertStmt | AssignStmt | Block | BranchStmt | CompFor | ConstDecl | DeferStmt |
+	EnumDecl | ExprStmt | FnDecl | ForCStmt | ForInStmt | ForStmt | GlobalDecl | GoStmt |
+	GotoLabel | GotoStmt | HashStmt | Import | InterfaceDecl | Module | Return | SqlStmt |
+	StructDecl | TypeDecl
+```
+
+All the AST struct declarations can be found in V source code: [vlib/v/ast/ast.v](https://github.com/vlang/v/blob/master/vlib/v/ast/ast.v)
+
+## AST tool
+
+If you are new to Vlang AST, You can install the [vast tool](https://github.com/lydiandy/vast). It can generate example code to AST json file.
+
+The json file can help you more understand the AST.
+
+```shell
+vast example.v       //generate example.json file and exit.
+
+vast -w example.v    //generate example.json and watch,if file change,regenerate.
```

Comment:
add space after `,`

Reply:
`// generate example.json, watch it, and regenerate if it changes.